src/test/java/org/jabref/logic/texparser/TexParserTest.java

```diff
@@ -0,0 +1,395 @@
+                    texResult.getUniqueKeys().get(EINSTEIN).get(1).getLineText());
+            assertEquals("\\cite{Darwin1888}.", texResult.getUniqueKeys().get(DARWIN).get(1).getLineText());
+
+            assertEquals("This is some content trying to cite a bib file: \\cite{Darwin1888}",
+                    texResult.getUniqueKeys().get(DARWIN).get(2).getLineText());
+            assertEquals("This is some content trying to cite a bib file: \\cite{Einstein1920}",
+                    texResult.getUniqueKeys().get(EINSTEIN).get(2).getLineText());
+            assertEquals("This is some content trying to cite a bib file: \\cite{Newton1999}",
+                    texResult.getUniqueKeys().get(NEWTON).get(0).getLineText());
+
+            // Check contexts
+            assertEquals("\\cite{Einstein1920}", texResult.getUniqueKeys().get(EINSTEIN).get(0).getContext());
+            assertEquals("cus, eu vehicula enim efficitur.~\\cite{Darwin1888}",
+                    texResult.getUniqueKeys().get(DARWIN).get(0).getContext());
+            assertEquals("Einstein said~\\cite{Einstein1920} that lorem impsu",
+                    texResult.getUniqueKeys().get(EINSTEIN).get(1).getContext());
+            assertEquals("\\cite{Darwin1888}.", texResult.getUniqueKeys().get(DARWIN).get(1).getContext());
+
+            assertEquals("ntent trying to cite a bib file: \\cite{Darwin1888}",
+                    texResult.getUniqueKeys().get(DARWIN).get(2).getContext());
+            assertEquals("ent trying to cite a bib file: \\cite{Einstein1920}",
+                    texResult.getUniqueKeys().get(EINSTEIN).get(2).getContext());
+            assertEquals("ntent trying to cite a bib file: \\cite{Newton1999}",
+                    texResult.getUniqueKeys().get(NEWTON).get(0).getContext());
+        }
+    }
+
+    @Test
+    void testDuplicateFiles() throws URISyntaxException, IOException {
+        final String DARWIN = "Darwin1888";
+        final String EINSTEIN = "Einstein1920";
+
+        InputStream originalStream = TexParserTest.class.getResourceAsStream("origin.bib");
+        Path texFile = Paths.get(TexParserTest.class.getResource("paper.tex").toURI());
+
+        try (InputStreamReader originalReader = new InputStreamReader(originalStream, StandardCharsets.UTF_8)) {
+            ParserResult result = new BibtexParser(importFormatPreferences,
+                    new DummyFileUpdateMonitor()).parse(originalReader);
+            TexParser texParser = new DefaultTexParser(result.getDatabase());
+            TexParserResult texResult = texParser.parse(Arrays.asList(texFile, texFile));
+            BibDatabase newDatabase = texResult.getGeneratedBibDatabase();
+
+            assertEquals(result.getDatabase(), texResult.getMasterDatabase());
+
+            // Check entries
+            assertTrue(texResult.getGeneratedBibDatabase().hasEntries());
```

Comment:
In general, it is better to create the expected result (here: `TexParserResult`) and then compare it in one `assertEquals` statement. To see why: change your implementation so that some of the tests fail, and then try to guess the source of the failure from the test output.